src/commands/flow/execute.ts

```diff
@@ -0,0 +1,596 @@
+import path from 'path'
+import { displayResult } from '../../utils/displayResult'
+import {
+  fileExists,
+  readFile,
+  isJsonFile,
+  isCsvFile,
+  createFile,
+  writeFile,
+  folderExists,
+  createFolder
+} from '../../utils/file-utils'
+import {
+  generateTimestamp,
+  parseLogLines,
+  millisecondsToDdHhMmSs,
+  asyncForEach
+} from '../../utils/utils'
+import { getAccessToken } from '../../utils/config-utils'
+import { Target } from '../../types'
+import SASjs from '@sasjs/adapter/node'
+import stringify from 'csv-stringify'
+import { setInterval } from 'timers'
+import examples from './examples'
+
+export async function execute(
+  source: string,
+  logFolder: string,
+  csvFile: string,
+  target: Target,
+  prefixAppLoc: Function
+) {
+  return new Promise(async (resolve, reject) => {
+    const commandExample = `sasjs flow execute --source /local/flow.json --logFolder /local/log/folder --csvFile /local/some.csv --target targetName`
+    const pollOptions = { MAX_POLL_COUNT: 24 * 60 * 60, POLL_INTERVAL: 1000 }
+
+    if (!source || !isJsonFile(source)) {
+      displayResult(
+        {},
+        `Please provide flow source (--source) file.\nCommand example: ${commandExample}`
+      )
+
+      reject(false)
+    }
+
+    if (!(await fileExists(source))) {
+      displayResult(
+        true,
+        `Source file does not exist.\nCommand example: ${commandExample}`
+      )
+
+      reject(false)
+    }
+
+    let sourceConfig = await readFile(source)
+
+    try {
+      sourceConfig = JSON.parse(sourceConfig)
+    } catch (error) {
+      throw `Invalid json file.`
+    }
+
+    let flows = sourceConfig.flows
```

Comment:
I'd suggest having slightly stricter validation for each `flow` object.
We can do that by having a `Flow` class with a static `parse` method which throws errors on format mismatch.
If the JSON is valid, it'll return a new `Flow` object.

Reply:
The class can have properties such as `predecessors` and `successors` which will give you auto-completion in the IDE as well.

Reply:
Good idea, I'll create a ticket for refactoring